reconcile/test/test_terraform_vpc_peerings.py

```diff
@@ -10,57 +13,333 @@
+                ocm.OCM.get_aws_infrastructure_access_terraform_assume_role
+            )
+        ocm_mock.get_aws_infrastructure_access_terraform_assume_role \
+            .side_effect = \
+            self.get_aws_infrastructure_access_terraform_assume_role
+        return ocm_mock
 
-class TestAWSAccountFromInfrastructureAccess(testslide.TestCase):
-    def setUp(self):
-        self.cluster = {
-            'name': 'cluster',
-            'spec': {
-                'region': 'region'
-            },
-            'network': {
-                'vpc': 'vpc'
-            },
-            'awsInfrastructureAccess': [
-                {
-                    'awsGroup': {
-                        'account': {
-                            'name': 'account',
-                            'uid': 'uid',
-                            'terraformUsername': 'terraform',
-                            'automationToken': 'token'
-                        }
+
+class MockAWSAPI:
+
+    def __init__(self) -> None:
+        self.vpc_details: dict[str, Tuple[str, list[str]]] = {}
+
+    def register(self, vpc: str,
+                 vpc_id: str, route_tables: list[str]) -> 'MockAWSAPI':
+        self.vpc_details[vpc] = (vpc_id, route_tables)
+        return self
+
+    def get_cluster_vpc_details(self, account: dict[str, Any],
+                                route_tables=False, subnets=False) -> Tuple:
+        if account["assume_cidr"] in self.vpc_details:
+            vpc_id, rt = self.vpc_details[account["assume_cidr"]]
+            if not route_tables:
+                return vpc_id, None, None
+            else:
+                return vpc_id, rt, None
+        else:
+            return None, None, None
+
+    def auto_speced_mock(self, mocker) -> aws_api.AWSApi:
+        aws_api_mock = mocker.patch(
+            "reconcile.utils.aws_api.AWSApi", autospec=True).return_value
+        aws_api_mock.get_cluster_vpc_details.mock_add_spec(
+            aws_api.AWSApi.get_cluster_vpc_details
+        )
+        aws_api_mock.get_cluster_vpc_details.side_effect = \
+            self.get_cluster_vpc_details
+        return aws_api_mock
+
+
+def build_cluster(name: str, vpc: Optional[str] = None,
+                  read_only_accounts: Optional[list[str]] = None,
+                  network_mgmt_accounts: Optional[list[str]] = None,
+                  peering_connections: Optional[list[dict[str, Any]]] = None):
+    if not vpc:
+        vpc = name
+    cluster = {
+        "name": name,
+        "spec": {
+            "region": "region"
+        },
+        "network": {
+            "vpc": vpc
+        },
+        "peering": {
+            "connections": peering_connections or []
+        },
+        "awsInfrastructureManagementAccounts": None
+    }
+
+    if read_only_accounts or network_mgmt_accounts:
+        cluster["awsInfrastructureManagementAccounts"] = []
+        if read_only_accounts:
+            for idx, acc in enumerate(read_only_accounts):
```

Comment:
Why are you enumerating here?

Reply:
apparently for no reason. i removed it in b28c9e7